rts/lra/test/basic/src/test/java/io/narayana/lra/arquillian/FailedLRAIT.java

```diff
@@ -106,7 +106,9 @@ static String getRecoveryUrl(URI lraId) {
     public void testWithStatusCompensateFailed() throws Exception {
         URI lraId = invokeInTransaction(LRAParticipantWithStatusURI.LRA_PARTICIPANT_PATH,
                 LRAParticipantWithStatusURI.TRANSACTIONAL_CANCEL_PATH, 500);
-
+        // Call the recoveryManager(recovery), after which we can remove "Thread.sleep"
+        // defaultPeriodicRecoveryInterval is 120 seconds
+        Thread.sleep(120000);
```

Comment:
In these tests, it is expected to get the deleted record immediately as LRA fails, but after my changes, the deleted records cannot be found until the recoveryManager runs. I am not sure how to proceed on this one. It would be great if you could point me to the right direction?

Reply:
It could be that the test expects the LRA being marked as failed. That's good and expected.
I think you should verify how the status call is invoked and why you are getting the different value than it was returned previously. I think maybe you need to merge the approaches - which may correspond with the race condition I mentioned above. The failed LRA should be considered based on the location in the object store (based on the `type`) but as well based on the LRA status when it has not been moved yet.

Reply:
ok